components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,358 @@
+        JSONObject json = new JSONObject();
+        json.put("id", this.id);
+        doReturn(this.id).when(this.patient).getId();
+        this.patientResource.updatePatient(json.toString(), this.id);
+
+        verify(this.logger).debug("Updating patient record [{}] via REST with JSON: {}", this.id, json.toString());
+    }
+
+    //----------------------------Delete Patient Tests----------------------------
+
+    @Test
+    public void deletePatientCantFindPatient() throws XWikiRestException {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+
+        Response response = this.patientResource.deletePatient(this.id);
+
+        verify(this.logger).debug("Patient record [{}] didn't exist", this.id);
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void deletePatientUserDoesNotHaveAccess() throws XWikiRestException {
+        doReturn(false).when(this.access).hasAccess(Right.DELETE, null, null);
+
+        Response response = this.patientResource.deletePatient(this.id);
+
+        verify(this.logger).debug("Delete access denied to user [{}] on patient record [{}]", this.currentUser, this.id);
+        Assert.assertEquals(Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void deletePatientCatchesXWikiException() throws XWikiRestException, XWikiException {
+        Provider provider = mock(Provider.class);
+        XWikiContext context = mock(XWikiContext.class);
+        XWiki wiki = mock(XWiki.class);
+        doReturn(context).when(provider).get();
+        doReturn(wiki).when(context).getWiki();
+        ReflectionUtils.setFieldValue(this.patientResource, "xcontextProvider", provider);
+
+        doReturn(true).when(this.access).hasAccess(Right.DELETE, null, null);
+
+        doThrow(XWikiException.class).when(wiki).deleteDocument(any(XWikiDocument.class), eq(context));
+
+        WebApplicationException ex = null;
+        try {
+            this.patientResource.deletePatient(this.id);
+        } catch(WebApplicationException temp) {
+            ex = temp;
+        }
+        Assert.assertNotNull("deletePatient did not throw a WebApplicationException as expected " +
+            "when catching an XWikiException", ex);
+        Assert.assertEquals(Status.INTERNAL_SERVER_ERROR.getStatusCode(), ex.getResponse().getStatus());
+        verify(this.logger).warn(eq("Failed to delete patient record [{}]: {}"), eq(this.id), anyString());
+    }
+
+    @Test
+    public void deletePatientReturnsNoContentResponse() throws XWikiRestException {
+        Provider provider = mock(Provider.class);
+        XWikiContext context = mock(XWikiContext.class);
+        XWiki wiki = mock(XWiki.class);
+        doReturn(context).when(provider).get();
+        doReturn(wiki).when(context).getWiki();
+        ReflectionUtils.setFieldValue(this.patientResource, "xcontextProvider", provider);
```

Comment:
Do you mean that it should be moved to the setUp method, or that this line doesn't need to exist at all?